// react-native-form-generator/src/fields/DatePickerField.ios.js
'use strict';

import React from 'react';
import ReactNative from 'react-native';
let { View, StyleSheet, TextInput, Text, PickerIOS, Dimensions} = ReactNative;
var {height, width}     = Dimensions.get('window');

import {DatePickerComponent} from '../lib/DatePickerComponent';

export class DatePickerField extends React.Component{
  setDate(date){
    this.refs.datePickerComponent.setDate(date);
  }
  render(){
    return(<DatePickerComponent
      {...this.props}
      ref='datePickerComponent'
      labelStyle={[formStyles.fieldText, this.props.labelStyle]}
      valueStyle = {[formStyles.fieldValue,this.props.valueStyle]}

      valueContainerStyle = {[formStyles.alignRight,
          formStyles.horizontalContainer,this.props.valueContainerStyle]}
      containerStyle={[
        formStyles.fieldContainer,
        formStyles.horizontalContainer,
        this.props.containerStyle,
      ]}
      />)
    }

  }

      let formStyles = StyleSheet.create({
        form:{
        
        },
        alignRight:{
           marginTop: 7, position:'absolute', right: 10
        },
        noBorder:{
          borderTopWidth: 0,
          borderBottomWidth: 0
        },
        separatorContainer:{
          // borderTopColor: '#C8C7CC',
          // borderTopWidth: 1,
          paddingTop: 35,
        },
        separator:{
          paddingLeft: 10,
          paddingRight: 10,
          color: "'rgba(0,43,55,1)'",
          paddingBottom: 7
        },
        fieldsWrapper:{
          // borderTopColor: '#afafaf',
          // borderTopWidth: 1,
        },
        horizontalContainer:{
          flexDirection: 'row',
          justifyContent: 'flex-start'
        },
        fieldContainer:{
          backgroundColor:'rgba(238, 232, 212, 1)',
          justifyContent: 'center',
          height: 45,
          flex: 1,
        },
        fieldValue:{
          fontSize: 34/2,
          paddingLeft: 10,
          paddingRight: 10,
          marginRight:10,
          paddingTop: 4,
          justifyContent: 'center',
          color: "'rgba(0,43,55,1)'",
        },
        fieldText:{
          fontSize: 34/2,
          paddingLeft: 10,
          paddingRight: 10,
          justifyContent: 'center',
          lineHeight: 32,
          color: "'rgba(0,43,55,1)'",
        },
        input:{
          paddingLeft: 10,
          paddingRight: 10,
          color: "'rgba(0,43,55,1)'",
        },
        helpTextContainer:{
          marginTop:9,
          marginBottom: 25,
          paddingLeft: 20,
          paddingRight: 20,
          color: "'rgba(0,43,55,1)'",
        },
        helpText:{
          color: '#7a7a7a',
        }
      });
