// react-native-form-generator/src/fields/PickerField.ios.js
'use strict';

import React from 'react';
import ReactNative from 'react-native';
let { View, StyleSheet, TextInput, Text, PickerIOS, Dimensions} = ReactNative;
var {height, width}     = Dimensions.get('window');


import {PickerComponent} from '../lib/PickerComponent';

export class PickerField extends React.Component{
  setValue(value){
    this.refs.fieldComponent.setValue(value)
  }
  render(){
    return(<PickerComponent
      {...this.props}
      ref='fieldComponent'
      labelStyle={[formStyles.fieldText, this.props.labelStyle]}
      valueStyle = {[formStyles.fieldValue,this.props.valueStyle]}
      valueContainerStyle = {[formStyles.alignRight,
          formStyles.horizontalContainer, this.props.valueContainerStyle]}
      containerStyle={[
        formStyles.fieldContainer,
        formStyles.horizontalContainer,
        this.props.containerStyle,
      ]}
      />)
    }

  }



      let formStyles = StyleSheet.create({
        form:{

        },
        alignRight:{
           marginTop:0, position:'absolute', right: 0
        },
        noBorder:{
          borderTopWidth: 0,
          borderBottomWidth: 0
        },
        separatorContainer:{
          // borderTopColor: '#C8C7CC',
          // borderTopWidth: 1,
          paddingTop: 35,
          borderBottomColor: '#C8C7CC',
          borderBottomWidth: 1,
          
        },
        separator:{

          paddingLeft: 10,
          paddingRight: 10,
          color: '#6D6D72',
          paddingBottom: 7

        },
        fieldsWrapper:{
          // borderTopColor: '#afafaf',
          // borderTopWidth: 1,
        },
        horizontalContainer:{
          flexDirection: 'row',
          justifyContent: 'flex-start',
        },
        fieldContainer:{
          backgroundColor: 'rgba(238, 232, 212, 1)',
          height: 55,
          alignItems: 'flex-start',
          justifyContent: 'center',
          width: width,
        },
        fieldValue:{
          fontSize: 14,
          paddingLeft: 12,
          paddingTop: 22,
          justifyContent: 'center',
          height: 55,
          width: width / 8 - 1,
          color: 'rgba(0,43,55,1)',
          alignItems: 'flex-end',
          borderLeftWidth:1,
          borderLeftColor: 'lightgray',
          backgroundColor: 'rgba(246, 240, 222, 1)',
        },
        fieldText:{
          fontSize: 18,
          paddingLeft: 10,
          paddingTop: 10,
          justifyContent: 'center',
          lineHeight: 32,
          color: 'rgba(0,43,55,1)',
        },
        input:{
          paddingLeft: 10,
          paddingRight: 10,
          backgroundColor:'white',
        },
        helpTextContainer:{
          marginTop:9,
          marginBottom: 25,
          paddingLeft: 20,
          paddingRight: 20,

        },
        helpText:{
          color: '#7a7a7a'
        }
      });
