// react-native-form-generator/src/fields/SwitchField.ios.js
'use strict';

import React from 'react';
let { View, StyleSheet, Text, Switch, Dimensions} = require('react-native');
var {height, width}     = Dimensions.get('window');
import {SwitchComponent} from '../lib/SwitchComponent';

export class SwitchField extends React.Component{
  setValue(value){
    this.refs.fieldComponent.setValue(value)
  }
  render(){

    return(<SwitchComponent
      {...this.props}
      ref='fieldComponent'
      containerStyle={[
        formStyles.fieldContainer,
        formStyles.horizontalContainer,
        this.props.containerStyle
      ]}

      labelStyle = {[
        formStyles.fieldText,
        this.props.labelStyle
      ]}
      switchStyle={[
        {marginTop: 7, position:'absolute', right: 10},
        this.props.switchStyle
      ]}
      />

    )
  }

}



  let formStyles = StyleSheet.create({
    fieldContainer:{
      backgroundColor: 'rgba(238, 232, 212, 1)',
      justifyContent: 'center',
      height: 45,
      flex: 1,
      alignItems: 'flex-start',
      marginRight: 10,
      marginLeft: 25,
      paddingLeft: 40,
    },
    horizontalContainer:{
      flexDirection: 'row',
      justifyContent: 'flex-start'
    },
    fieldText:{
      fontSize: 34/2,
      paddingLeft: 10,
      paddingRight: 10,
      justifyContent: 'center',
      lineHeight: 32
    },

  });
